src/services/CapabilitiesManager.ts

```diff
@@ -41,14 +57,42 @@ export function hasTalkFeature(token: string = 'local', feature: string): boolea
  * @param key2 second-level key (e.g. 'allowed')
  */
 export function getTalkConfig(token: string = 'local', key1: keyof Config, key2: keyof Config[keyof Config]) {
+	const remoteCapabilities = getRemoteCapability(token)
 	if (localCapabilities?.spreed?.['config-local']?.[key1]?.includes(key2)) {
 		return localCapabilities?.spreed?.config?.[key1]?.[key2]
-	} else if (token === 'local' || !remoteCapabilities[token]) {
+	} else if (token === 'local' || !remoteCapabilities) {
 		return localCapabilities?.spreed?.config?.[key1]?.[key2]
 	} else {
 		// TODO discuss handling remote config (respect remote only / both / minimal)
-		return remoteCapabilities[token]?.spreed?.config?.[key1]?.[key2]
+		return remoteCapabilities?.spreed?.config?.[key1]?.[key2]
+	}
+}
+
+/**
+ * Returns capability for specified token (if already matches from one of remote servers)
+ * @param token token of the conversation
+ */
+function getRemoteCapability(token: string): RemoteCapability | null {
+	if (remoteCapabilities[remoteTokenMap[token]]) {
+		return remoteCapabilities[remoteTokenMap[token]]
+	}
+
+	const cachedConversations = BrowserStorage.getItem('cachedConversations')
+	if (!cachedConversations?.length) {
+		return null
 	}
+
+	const remoteServer = JSON.parse(cachedConversations)?.[token]?.remoteServer
+
+	if (remoteServer && remoteCapabilities[remoteServer]) {
+		console.debug(`Reuse remote capabilities from another conversation (same remote server ${remoteServer})`)
+		remoteCapabilities[remoteServer].tokens = [...new Set((remoteCapabilities[remoteServer].tokens || []).concat(token))]
+		BrowserStorage.setItem('remoteCapabilities', JSON.stringify(remoteCapabilities))
+		remoteTokenMap = generateTokenMap()
+		return remoteCapabilities[remoteServer]
+	}
```

Comment:
Given that we searched in cached conversations using the current conversation token, we are basically looking in the same conversation, it is almost equal to 
```
if (remoteCapabilities[remoteTokenMap[token]]) {
		return remoteCapabilities[remoteTokenMap[token]]
	}
```
except that it is guaranteed to have server name from cached conversations.
if server ever existed from another conversation, it will be present in `remoteCapabilities` anyway

Reply:
For updating token in tokens, maybe use simple existence check before appending the item? and we don't need to generate the map again as we can only add the new token.

Reply:
reworked creating of remoteTokenMap, should be appropriate now